distributed/shuffle/_merge.py

```diff
@@ -213,9 +215,7 @@ def __init__(
         self.n_partitions_right = n_partitions_right
         self.left_index = left_index
         self.right_index = right_index
-        annotations = annotations or {}
-        annotations.update({"shuffle": lambda key: key[-1]})
-        super().__init__(annotations=annotations)
```

Comment:
I suggest to still keep the annotations. They will not hurt and we might want to utilize them *iff* they are available.